keras/backend/cntk_backend.py

```diff
@@ -1310,7 +1310,15 @@ def rnn(step_function, inputs, initial_states,
             initial.append(s)
 
     need_convert = not has_seq_axis(inputs)
+    if go_backwards and need_convert is False:
+        raise NotImplementedError('CNTK Backend: reverse ordered rnn with '
```

Comment:
The error message seems unclear. Please update it to something like:
"`go_backards` is not support with variable-length sequences. Please specify a static-length for your sequences"